packages/react/react/src/hooks/useCombinedRefs.ts

```diff
@@ -18,17 +16,15 @@ type CallbackRef<T> = (ref: T | null) => void;
  *   return <div ref={ref} />;
  * })
  */
-export function useCombinedRefs<T>(...refs: Array<Ref<T> | CallbackRef<T>>): Ref<T> {
-  return useCallback(
-    (value: T) => {
-      for (const ref of refs) {
+export const useCombinedRefs = <T>(...refs: Array<Ref<T>>): RefCallback<T> =>
+  useCallback(
+    (value: T) =>
+      refs.forEach(ref => {
         if (typeof ref === 'function') {
           ref(value);
         } else if (ref != null) {
           (ref as MutableRefObject<T>).current = value;
         }
-      }
-    },
+      }),
     [refs]
```

Comment:
@tooooo1 
refs will be always new array. so I spread it to optimize correctly.
and then I added eslint-disable-next-line.

chakra-ui do this to optimize correctly too. you can check it in [@chakra-ui's useMergeRefs](https://github.com/chakra-ui/chakra-ui/blob/main/packages/hooks/use-merge-refs/src/index.ts)

```suggestion
    // eslint-disable-next-line react-hooks/exhaustive-deps
    refs
```

Reply:
Strictly, I think this is not optimization. I think optimization is improving performance of something.

With or without your suggestion, there's not any performance improvement of `useCombinedRefs`.

But, with your suggestion, the potential that the reference of the returned callback of `useCombinedRefs` changed on every render will be disappeared and developers won't have to memoize the `refs` parameter.